AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -247,48 +214,96 @@ - (void)readWithPartition:(NSString *)partition
                    }];
 }
 
+- (void)listWithPartition:(NSString *)partition
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *_Nullable)readOptions
+        continuationToken:(nullable NSString *)continuationToken
+        completionHandler:(MSPaginatedDocumentsCompletionHandler)completionHandler {
+  NSMutableDictionary *additionalHeaders = [NSMutableDictionary new];
+  if (continuationToken) {
+    [additionalHeaders setObject:(NSString *)continuationToken forKey:kMSDocumentContinuationTokenHeaderKey];
+  }
+  // TODO: Add constant and set max items (maybe provide a way to change it e.g. in the read options)
+  [additionalHeaders setObject:@"10" forKey:@"x-ms-max-item-count"];
+
+  // Call cosmos DB.
+  [self performOperationForPartition:partition
+                          documentId:nil
+                          httpMethod:kMSHttpMethodGet
+                                body:nil
+                   additionalHeaders:additionalHeaders
+        completionHandlerWithHeaders:^(NSData *_Nullable data, NSDictionary *_Nullable headers, NSError *_Nonnull cosmosDbError) {
+          // If not OK.
+          if (!data || [MSDataSourceError errorCodeFromError:cosmosDbError] != kMSACDocumentSucceededErrorCode) {
+            MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@", [cosmosDbError description]);
+            MSDataSourceError *dataSourceCosmosDbError = [[MSDataSourceError alloc] initWithError:cosmosDbError];
+            MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceCosmosDbError];
+            completionHandler(documents);
+            return;
+          }
+
+          // Deserialize.
+          NSError *deserializeError;
+          NSDictionary *json = [NSJSONSerialization JSONObjectWithData:(NSData *)data options:0 error:&deserializeError];
+          if (deserializeError) {
+            MSLogError([MSDataStore logTag], @"Error deserializing data: %@", [deserializeError description]);
+            MSDataSourceError *dataSourceDeserializeError = [[MSDataSourceError alloc] initWithError:deserializeError];
+            MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceDeserializeError];
+            completionHandler(documents);
+            return;
+          }
+          MSLogDebug([MSDataStore logTag], @"Documents json: %@", json);
```

Comment:
Customers sometimes copy paste console logs to GitHub issues, if the document contains sensible info it could be problematic. Do we still want to print it?